### ISSUE TITLE
Error in Implementation

### ISSUE THREAD

**Issue by AS-researcher6** [neale/Adversarial-Autoencoder#3]
The decoder and discriminator are being treated as a GAN, which is incorrect in the implementation of an adversarial autoencoder. In your code, you train the discriminator to distinguish between real images and those generated by the decoder from random noise.
However, in an adversarial autoencoder, the encoder and discriminator are the modules that are adversarially trained. The discriminator should be distinguishing between gaussian noise and real latent space embeddings of the encoder in order to constrain p(z|x) to be gaussian. This explains why your samples are so excellent, while your reconstructions are not.
See Figure 1 of the Adversarial Autoencoders paper for a visual of the proper architecture: https://arxiv.org/pdf/1511.05644.pdf

**Comment by neale** [comment]
Right, I should make this clear in the readme. I built more of an autoencoder-GAN than a true adversarial autoencoder -- this has better results if you want random samples more than reconstructions, which is often more desirable. 
Maybe this week I'll add a branch with a true adversarial autoencoder. I'll leave this issue open until I do that. Also, it looks like PyTorch fixed [#19024](https://github.com/pytorch/pytorch/issues/19024#event-2489479809) so I have some work to do with the multi-gpu code anyway.
